### PR TITLE
Refactor tizen surface

### PR DESCRIPTION
**Pull request by bbrto21**
* Add TizenNativeWindow, TizenNativeEGLWindow, TizenWl2Display
* Add TizenEGLContext, TizenEGLSurface, TizenEGLSurface
* TizenEmbedderEngine has TizenNativeWindow as a public member
* Release EGLSurface, EGLDisplay, EGLContext

Signed-off-by: Boram Bae <boram21.bae@samsung.com>

### CODE REVIEW

**Comment by bbrto21** [comment]
I am trying to fix bugs related to releasing resources. I am also trying to re-implement a rotation(or OnTizenSurfaceResize). 
As far as I know, for the input panel(software keyboard) to be positioned properly according to the device orientation, the native window must be rotated by itself.
Anyone who knows about this, please let me know(I am not familiar with this field.)

**Review by bwikbs — APPROVED** [review]

File: shell/platform/tizen/tizen_native_window.cc
Comment (seungsoo47) [inline shell/platform/tizen/tizen_native_window.cc:13]: Don't you need abort()?

Comment (bbrto21) [reply shell/platform/tizen/tizen_native_window.cc:13]: I fixed it

**Review by seungsoo47 — APPROVED** [review]
LGTM

File: shell/platform/tizen/tizen_event_loop.cc
Comment (bbrto21) [inline shell/platform/tizen/tizen_event_loop.cc:21]: These are my mistakes, I'll clean up soon, Please do not merge

File: shell/platform/tizen/tizen_native_window.cc
Comment (swift-kim) [inline shell/platform/tizen/tizen_native_window.cc:54]: Please follow the naming convention for variables.
https://google.github.io/styleguide/cppguide.html#Variable_Names

File: shell/platform/tizen/tizen_native_window.cc
Comment (swift-kim) [inline shell/platform/tizen/tizen_native_window.cc:97]: Same here and in `TizenSurfaceGL`.

File: shell/platform/tizen/tizen_surface_gl.cc
Comment (swift-kim) [inline shell/platform/tizen/tizen_surface_gl.cc:243]: Change the log level.

**Review by swift-kim — COMMENTED** [review]
👍

Please add some comments on the classes or explain the overall design when you have time.

File: shell/platform/tizen/tizen_surface_gl.cc
Comment (bwikbs) [reply shell/platform/tizen/tizen_surface_gl.cc:230]: Hm? I'm not sure .. but why? what's wired?

Comment (swift-kim) [reply shell/platform/tizen/tizen_surface_gl.cc:230]: The file `proc_resolver_macro.h` is _incomplete_ by itself and implements only a part of a specific function in another file. I don't find it's quite natural. Why don't you create a static utility function that takes name as a parameter and then use it in `OnProcResolver`?

Comment (bbrto21) [reply shell/platform/tizen/tizen_surface_gl.cc:230]: I made only minimal effort, but I think it is nothing special that including header file that has partial implementation or large string or only macros or something else.

File: shell/platform/tizen/tizen_native_window.cc
Comment (bbrto21) [reply shell/platform/tizen/tizen_native_window.cc:15]: I think that If ecore_wl2_init fails, nothing can be done normally so abort is not bad solution.
Of course, since abort means an abnormal termination, I think it is far from safety.
if you have any suggestion please let' me know feel free. anyway I'll remove return

File: shell/platform/tizen/tizen_event_loop.cc
Comment (bbrto21) [reply shell/platform/tizen/tizen_event_loop.cc]: I fixed it

File: shell/platform/tizen/tizen_native_window.cc
Comment (bbrto21) [reply shell/platform/tizen/tizen_native_window.cc]: I fixed it

File: shell/platform/tizen/tizen_surface_gl.cc
Comment (bbrto21) [reply shell/platform/tizen/tizen_surface_gl.cc]: I fixed it

File: shell/platform/tizen/tizen_surface_gl.cc
Comment (bbrto21) [reply shell/platform/tizen/tizen_surface_gl.cc]: I just reverted the changes

**Review by swift-kim — APPROVED** [review]

File: shell/platform/tizen/tizen_native_window.cc
Comment (bbrto21) [reply shell/platform/tizen/tizen_native_window.cc:14]: We decided to keep the changes we initially proposed after ofline-discussion.
This case will eventually be handled by app-host.

**Comment by bbrto21** [comment]
This PR should be merged after #12 is merged.

**Pull request merged**
